### DISCUSSION TITLE
dvtws на FreeBSD 7.4, проблема с ipfw sockarg

### DISCUSSION THREAD

**Discussion by Max3652** [bol-van/zapret#270] [Q&A]
dvtws собралась на FreeBSD 7.4, проблема с опцией ipfw sockarg, она не поддерживается. 
Как написано в bsd.txt, опция sockarg используется для предотвращения зацикливания.
ipfw add 100 divert 989 tcp from any to any 80,443 out not diverted not sockarg xmit em0
Если no sockarg убрать, то ютуб прекрасно работает.
Есть ли другой способ предотвращения зацикливания в варианте с dvtws+ipfw?

**Comment by bol-van** [comment]
в ближайшее время планирую отказ от сокарга и перевод всего отсыла на диверт
7.4 это что-то из нулевых

**Comment by nuclight** [comment]
> Есть ли другой способ предотвращения зацикливания в варианте с dvtws+ipfw?

Можно попробовать выделить демону отдельный uid и соответственно вписать в правило соответствующий `not uid`

Так-то конечно полный перевод всего на divert правильнее, специально созданный механизм.

**Comment by bol-van** [comment]
сидеть в 2024 году на ос 2008 года - вот главная проблема

**Reply by nuclight** [reply]
Ну мало ли зачем, вот у меня лет десять валяется первопень с как раз этой версией, я его думал приспособить под роутер как раз, но всё лениво было...
